constrain-lib/src/main/resources/schema/dependency-constraints-schema.json

```
@@ -0,0 +1,137 @@
+          "examples": [
+            "com.example",
+            "org.springframework"
+          ]
+        },
+        "name": {
+          "type": "string",
+          "title": "Name",
+          "description": "The name or artifact id of the dependency.",
+          "examples": [
+            "example-dependency",
+            "spring-core"
+          ]
+        },
+        "suggestedVersion": {
+          "type": "string",
+          "title": "Suggested Version",
+          "description": "The suggested version of the dependency. This is needed because, at it's core, the Gradle Dependency Engine is a constraints solver. However, without knowledge of a known version that is potentially safe, the constraint will simply cause the Gradle build to fail. As such, a suggested version is used to provide a hint to the constraint engine what version it may want to use. \n\n IMPORTANT NOTE: When used in the context of security updates, the suggested version should be above all currently resolved versions of this dependency while simultaneously the lowest version of the dependency that is considered safe.",
+          "examples": [
+            "1.0.0"
+          ]
+        },
+        "rejectedVersions": {
+          "type": "array",
+          "title": "Rejected Versions",
+          "description": "The list of versions that are rejected by this constraint.",
+          "items": {
+            "type": "string",
+            "title": "Rejected Version",
+            "description": "This supports either specific versions or the Gradle version range format: https://docs.gradle.org/current/userguide/single_versions.html",
+            "examples": [
+              "1.0.0",
+              "[1.0,1.4]",
+              "[1.0,1.4)"
+            ]
+          }
+        },
+        "because": {
+          "type": "object",
+          "title": "Because",
+          "description": "Will be used to generate the Dependency Constraint 'because' reason. The reason will contain the `advisoryIdentifiers` as a prefix, with the `reason` as the suffix.\nExample: '[CVE-2019-1234, GHSA-aaaa-bbbb-cccc]: Contains an SQL injection vulnerability.",
+          "properties": {
+            "reason": {
+              "type": "string",
+              "title": "Reason",
+              "description": "The reason why this constraint is being applied.",
+              "examples": [
+                "Contains an SQL injection vulnerability."
+              ]
+            },
+            "advisoryIdentifiers": {
+              "type": "array",
+              "title": "Advisory Identifiers",
+              "description": "The list of advisory identifiers that are associated with this constraint.",
+              "items": {
+                "type": "string",
+                "title": "Advisory Identifier",
+                "description": "The advisory identifier.",
+                "examples": [
+                  "CVE-2019-1234",
+                  "GHSA-aaaa-bbbb-cccc"
```

It'd be nice for these examples to all be either real advisories or invalid advisories. Currently, the CVE example [is a real vulnerability](https://nvd.nist.gov/vuln/detail/CVE-2019-1234), whereas GHSA isn't. Unfortunately, GHSA's are randomly generated, so there's no guarantees.

---

Here's a suggestion if you'd like to use real identifiers:

```suggestion
                  "CVE-2019-1234",
                  "GHSA-2xwp-m7mq-7q3r"
```

---

...and here's a suggestion if you'd prefer fake identifiers:

```suggestion
                  "CVE-2019-1234",
                  "GHSA-5qhq-9w8w-49q6"
```